tests/mtouch/ToolTasksBinPathTest.cs

```diff
@@ -82,6 +84,7 @@ public void CheckToolBinDir (string taskName, string binDirToCheck)
 				RedirectStandardError = true,
 			};
 			psi.EnvironmentVariables.Add ("DEVELOPER_DIR", Configuration.xcode_root);
+			psi.EnvironmentVariables.Remove ("XCODE_DEVELOPER_DIR_PATH"); // VSfM sets XCODE_DEVELOPER_DIR_PATH, which confuses the command-line tools if it doesn't match xcode-select, so just unset it.
```

Comment:
that change is not mentioned in the commit message

Reply:
Wouldn't that cause problems in VSM? We need to make sure it resets it if needed or maybe tell the VSM team to not set it at all.

Reply:
This problem occurs when running tests from within VSfM, and those tests inherit VSfM's environment, and this variable causes trouble when running Apple's command line tools if it doesn't match what xcode-select says - it will never be a problem for VSfM because that environment is not affected by this change.

It's already filed here: https://github.com/xamarin/xamarin-macios/issues/3931

Reply:
PR description has been updated.